Tests/MessagingInApp/Gist/Managers/MessageQueueManagerTest.swift

```diff
@@ -64,6 +64,45 @@ class MessageQueueManagerTest: UnitTest {
         XCTAssertEqual(actualMessages, [givenMessage2, givenMessage1, givenMessage3])
     }
 
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMatch_expectInlineMessage() {
+        let givenElementId1 = String.random
+        let givenElementId2 = String.random
+        let givenElementId3 = String.random
+        let currentRoute = String.random
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId1, priority: 1)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId2, priority: 0)
+        let givenMessage3 = Message(elementId: givenElementId3, priority: 0)
```

Comment:
For the `Message pageRule`, I don't recommend using a `String.random` value. A `Message` page rule is [a regex string](https://github.com/customerio/customerio-ios/blob/edcd7765e3388ba4d16424a2c798149d09b4b15c/Sources/MessagingInApp/Gist/Managers/Models/Message.swift#L118-L127). To make the test more accurate, I suggest using a regex pattern for `Message(pageRule)` parameter.

Reply:
From reading the test function name, it looks like the test is scoped to only test the page rules. I agree with that scope. 

To keep this test more focused on just testing page rules, I suggest making the element id and priorities of all of the given messages to be the same. So the test is only testing 1 Message property instead of 3 separate ones.

Reply:
Correct, it has been scoped to test only page rules. To your suggestion to make all element Ids and priorities same, the test would look something like:

```
let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 0)
let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 0)
let givenMessage3 = Message(elementId: givenElementId, priority: 0)
```

Since all of of same priority then ordering can be random, that might end up test cases to fail at times. Instead I have updated the test case to have same element ids but different priorities.

Reply:
Good point about the randomness. 

I also encountered that and implemented a fix [in this PR](https://github.com/customerio/customerio-ios/pull/756/files#r1661299354). 

To not block this PR, I am OK with have same element ids and different priorities. If you would rather change the base branch of your PR to `levi/is-there-a-message-next-in-queue-after-close-button` so you can re-use that bug fix, I would be OK with that, too.

Reply:
Updated the base branch to `levi/is-there-a-message-next-in-queue-after-close-button`

Reply:
Very sorry to complicate this, Aman. 

I am making some modifications to the `levi/is-there-a-message-next-in-queue-after-close-button` branch to [prevent some bugs I discovered](https://linear.app/customerio/issue/MBL-313/after-close-action-pressed-replace-in-app-message-with-next-in-queue#comment-68684ba6). 

Part of these modifications is reverting the fix I suggested in this thread. 

Therefore, I suggest moving forward with your original plan: "updated the test case to have same element ids but different priorities."

Reply:
No worries, Levi ! Thanks for letting me know.